plugins/pulp_docker/plugins/distributors/configuration.py

```diff
@@ -324,7 +324,9 @@ def _is_valid_repo_registry_id(repo_registry_id):
     :return:                 True if valid, False if invalid
     :rtype:                  boolean
     """
-
-    component_re = re.compile('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+    # https://github.com/docker/distribution/blob/master/reference/regexp.go#L18
+    # the Go playground was used https://play.golang.org/p/wQ4w431jvS which lead
+    # to the following output:
+    component_re = re.compile('[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?$')
```

Comment:
i did not want to manipulate with this at all so i took the output from the playground as it was, in case docker changes something we'd just replace with new regex

Reply:
I suggest to add `^` at the beginning to explicitly specify that the intent is to have a full match.
I believe it works as it is now because you use `re.match`, I'm not sure it will work with `re.search`, for example.

So +1 to make it `re.compile(r'^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?$')`